pkg/cmd/server/admin/create_nodeconfig.go

```diff
@@ -0,0 +1,375 @@
+	ServerKeyFile   string
+	APIServerCAFile string
+	APIServerURL    string
+}
+
+func NewCommandNodeConfig() *cobra.Command {
+	options := NewDefaultCreateNodeConfigOptions()
+
+	cmd := &cobra.Command{
+		Use:   "create-node-config",
+		Short: "Create a portable client folder containing a client certificate, a client key, a server certificate authority, and a .kubeconfig file.",
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Validate(args); err != nil {
+				fmt.Println(err.Error())
+				c.Help()
+				return
+			}
+
+			if err := options.CreateNodeFolder(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+
+	BindGetSignerCertOptions(options.GetSignerCertOptions, flags, "")
+
+	flags.StringVar(&options.NodeConfigDir, "node-dir", "", "The client data directory.")
+
+	flags.StringVar(&options.NodeName, "node", "", "The name of the node as it appears in etcd.")
+	flags.Var(&options.Hostnames, "hostnames", "Every hostname or IP you want server certs to be valid for. Comma delimited list")
+	flags.StringVar(&options.VolumeDir, "volume-dir", options.VolumeDir, "The volume storage directory.  This path is not relativized.")
+	flags.StringVar(&options.NetworkContainerImage, "network-container-image", options.NetworkContainerImage, "The exact name of the image.  No processing is done on this argument.")
+	flags.BoolVar(&options.AllowDisabledDocker, "allow-disabled-docker", options.AllowDisabledDocker, "Allow the node to start without docker being available.")
+	flags.StringVar(&options.DNSDomain, "dns-domain", options.DNSDomain, "DNS domain for the cluster.")
+	flags.StringVar(&options.DNSIP, "dns-ip", options.DNSIP, "DNS server IP for the cluster.")
+	flags.Var(&options.ListenAddr, "listen", "The address to listen for connections on (scheme://host:port).")
+
+	flags.StringVar(&options.ClientCertFile, "client-certificate", "", "The client cert file.")
+	flags.StringVar(&options.ClientKeyFile, "client-key", "", "The client key file.")
+	flags.StringVar(&options.ServerCertFile, "server-certificate", "", "The server cert file for serving secure traffic.")
+	flags.StringVar(&options.ServerKeyFile, "server-key", "", "The server key file for serving secure traffic.")
+	flags.StringVar(&options.APIServerURL, "master", options.APIServerURL, "The API server's URL.")
+	flags.StringVar(&options.APIServerCAFile, "certificate-authority", options.APIServerCAFile, "Path to the API server's CA file.")
+
+	return cmd
+}
+
+func NewDefaultCreateNodeConfigOptions() *CreateNodeConfigOptions {
+	options := &CreateNodeConfigOptions{GetSignerCertOptions: &GetSignerCertOptions{}}
+	options.VolumeDir = "openshift.local.volumes"
+	options.DNSDomain = "local"
+	options.APIServerURL = "https://localhost:8443"
+	options.APIServerCAFile = "openshift.local.certificates/ca/cert.crt"
+
+	imageTemplate := variable.NewDefaultImageTemplate()
+	options.NetworkContainerImage = imageTemplate.ExpandOrDie("pod")
+
+	options.DNSIP = "127.0.0.1"
```

Comment:
do we want to default to the IP of the machine you are running on, or leave DNSIP blank by default (so it resolves based on the service IP?)